freestyle/shared/src/main/scala/freestyle/free.scala

```diff
@@ -4,6 +4,10 @@ import scala.annotation.{compileTimeOnly, StaticAnnotation}
 import scala.language.experimental.macros
 import scala.reflect.macros.blackbox
 
+trait FreeModuleLike {
```

Comment:
For now I would just leave it as an exclusive concern of the `@macro` implementation. I prefer not to put more _meaningful_ words, for now.